src/engine.cc

```diff
@@ -162,7 +165,7 @@ SearchLimits EngineController::PopulateSearchLimits(int ply, bool is_black,
   // Make sure we don't exceed current time limit with what we calculated.
   limits.time_ms = std::max(
       int64_t{0},
-      std::min(static_cast<int64_t>(this_move_time), time - move_overhead));
+      std::min(static_cast<int64_t>(this_move_time), time - lagbuffer - move_overhead));
```

Comment:
Again if it is working as a "worst case" buffer then I think this line just needs to be `time - max(lagbuffer, move_overhead)`

Reply:
Good call using max() instead.